src/chat-handler.js

```diff
@@ -0,0 +1,28 @@
+
+const chatHandler = (route, data) => {
+  switch (route) {
+    case ('CHATROOM.REGISTER') : {
+      newuser = new NewUser(data);
+      newuser.isValid();
+      break;
+    }
+
+    case ('CHATROOM.MESSAGE') : {
+      io.emit('message', data);
+      break;
+    }
+
+    case ('CHATROOM.DISCONNECT') : {
+      // user close window
+      break;
+    }
+  }
+};
+
+io.on('connection', (socket) => {
+  socket.on('message', (message) => {
```

Comment:
Awesome! Perhaps on 'disconnect', a message is sent into the above switch statement with 'chat.disconnect' message? Then the clientid will get removed from the room?